pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -733,11 +733,23 @@ func getNodeOrPVCName(d *appsv1.Deployment) (string, error) {
 	if v, ok := d.Labels[OSDOverPVCLabelKey]; ok {
 		return v, nil // OSD is on PVC
 	}
+	for k, v := range d.Spec.Template.Spec.NodeSelector {
+		if k == k8sutil.LabelHostname() {
+			return v, nil
+		}
+	}
+	// try to fallback on default hostname label.
+	// needed to perform upgrde to custom hostname node label:
+	// question: how to handle custom label update? We need previous label name here.
+	// in case of non-portable osd NodeSelector always have one entry and it is easy to get previous label.
+	// but how to handle portable OSD?
+	// https://github.com/rook/rook/blob/a9af67d68532b5686c588f0f7c650e24df2432ac/pkg/operator/ceph/cluster/osd/spec.go#L754-L761
```

Comment:
[Clickable link](https://github.com/rook/rook/blob/a9af67d68532b5686c588f0f7c650e24df2432ac/pkg/operator/ceph/cluster/osd/spec.go#L754-L761). 

This logic is required to upgrade existing clusters to a new label. To do upgrade it needs to know previous value of hostname label. It is trivial to upgrade from `kubernetes.io/hostname` or if `NodeSelector` contains a single entry.  But for portable osd (see link) node selector is complex and it is not clear how to handle it.

Reply:
The portable OSDs won't have a node selector. They just use the node affinity or other placement settings, then K8s assigns it to a node. So the node selector should remain empty for portable OSDs.